pegasus_wrapper/workflow.py

```diff
@@ -277,6 +324,159 @@ def run_python_on_parameters(
+            params_path=job_dir / "____params.params",
+            stdout_file=Path(stdout_path),
+            ckpt_path=checkpoint_path,
+            override_conda_config=override_conda_config,
+            python="pypy3" if use_pypy else "python",
+            treat_params_as_cmd_args=True,
+            pre_job=pre_job_bash,
+            post_job=post_job_bash,
+        )
+        if (
+            self._job_name_for(job_name)
+            not in self._transformation_name_to_transformation
+        ):
+            logging.info(
+                "Transformation %s recognized as new", self._job_name_for(job_name)
+            )
+            self._transformation_name_to_transformation[
+                self._job_name_for(job_name)
+            ] = Transformation(
+                self._job_name_for(job_name),
+                namespace=self._namespace,
+                version="4.0",
+                site=self._default_site,
+                pfn=str(script_path.absolute()),
+                is_stageable=job_is_stageable,
+                bypass_staging=job_bypass_staging,
+                arch=Arch.X86_64,
+                os_type=OS.LINUX,
+            )
+            self._transformation_catalog.add_transformations(
+                self._transformation_name_to_transformation[self._job_name_for(job_name)]
+            )
+
+        script_executable = self._transformation_name_to_transformation[
+            self._job_name_for(job_name)
+        ]
+
+        job = Job(script_executable)
+
+        set_pegasus_args = set_pegasus_args.split(" ")
+
+        args: List[Union[File, float, int, str]] = []
+        for tok in set_pegasus_args:
+            if tok in self._lfn_to_file:
+                args.append(self._lfn_to_file[tok])
+            else:
+                args.append(tok)
+
+        job.add_args(*args)
+
+        self._job_graph.add_jobs(job)
+        for parent_dependency in depends_on:
+            if parent_dependency.job:
+                self._job_graph.add_dependency(job, parents=[parent_dependency.job])
+            for out_file in parent_dependency.output_files:
+                job.add_inputs(out_file)
+
+        resource_request = self.set_resource_request(resource_request)
+
+        if category:
+            job.add_dagman_profile(category=category, retry=times_to_retry_job)
+
+        resource_request.apply_to_job(job, job_name=self._job_name_for(job_name))
+
+        # Handle Output Files
+        # This is currently only handled as the checkpoint file
+        # See: https://github.com/isi-vista/vista-pegasus-wrapper/issues/25
+        # If the checkpoint file already exists, we want to add it to the replica catalog
+        # so that we don't run the job corresponding to the checkpoint file again
+        checkpoint_pegasus_file = self.create_file(
+            f"{ckpt_name}", checkpoint_path, add_to_catalog=checkpoint_path.exists()
+        )
+
+        job.add_checkpoint(checkpoint_pegasus_file, stage_out=False)
```

Comment:
When testing this, do the jobs properly checkpoint with the `stage_out=False`? Haven't gotten around to checking out if the change to `add_checkpoint` was enough to do the trick or if we still need to figure something out with the staging errors.

Reply:
Need to do more robust testing to verify -- Cluster was doing maintenance when I was finalizing cleaning up this code.